src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -454,14 +461,10 @@ internal sealed class RSession : IRSession, IRCallbacks {
         /// Graph app may call Win32 API directly rather than going via R API callbacks.
         /// </summary>
         /// <returns>Pressed button code</returns>
-        async Task<MessageButtons> IRCallbacks.ShowDialog(IReadOnlyList<IRContext> contexts, string s, bool isEvaluationAllowed, MessageButtons buttons, CancellationToken ct) {
+        async Task<MessageButtons> IRCallbacks.ShowDialog(IReadOnlyList<IRContext> contexts, string s, MessageButtons buttons, CancellationToken ct) {
```

Comment:
Yes, in a sense that you can call `EvaluateAsync` at any time now. There's still a queue on the host side, and if there's an active eval that is not interruptible/reentrant, other evals will not be run in the middle of it. But they will not fail, either - they'll just stay in the queue, and run at the next available opportunity.